torchaudio/functional/functional.py

```diff
@@ -135,6 +136,95 @@ def spectrogram(
+            dimension (..., freq, time, 2), if return_complex is False, although this
+            behavior is deprecated.
+        length (int, optional): The amount to trim the signal by (i.e. the original signal length).
+            Default: None (whole signal).
+        pad (int): Two sided padding of signal
+        window (Tensor): Window tensor that is applied/multiplied to each frame/window
+        n_fft (int): Size of FFT
+        hop_length (int): Length of hop between STFT windows
+        win_length (int): Window size
+        normalized (bool): Whether the stft output was normalized by magnitude
+        center (bool, optional): whether the waveform was padded on both sides so
+            that the :math:`t`-th frame is centered at time :math:`t \times \text{hop\_length}`.
+            Default: ``True``
+        pad_mode (string, optional): controls the padding method used when
+            :attr:`center` is ``True``. This parameter is provided for compatibility with the
+            spectrogram function and is not used. Default: ``"reflect"``
+        onesided (bool, optional): controls whether spectrogram was done in onesided mode.
+            Default: ``True``
+        return_complex (bool, optional):
+            This value is ignored, and a real-valued output is always returned. It is provided only
+            for compatibility with the spectrogram function.
+            Default: ``False``
+
+    Returns:
+        Tensor: Dimension (..., time). Least squares estimation of the original signal.
+    """
+
+    if spectrogram.dtype == torch.float32 or spectrogram.dtype == torch.float64:
+        warnings.warn(
+            "The use of pseudo complex type in inverse_spectrogram is now deprecated. "
+            "Please migrate to native complex type by using a complex tensor as input. "
+            "If the input is generated via spectrogram() function or transform, please use "
+            "return_complex=True as an argument to that function. "
+            "Please refer to https://github.com/pytorch/audio/issues/1337 "
+            "for more details about torchaudio's plan to migrate to native complex type."
+        )
+        spectrogram = torch.view_as_complex(spectrogram)
+
+    if normalized:
+        spectrogram = spectrogram * window.pow(2.).sum().sqrt()
+
+    # pack batch
+    shape = spectrogram.size()
+    spectrogram = spectrogram.reshape(-1, shape[-2], shape[-1])
+
+    # default values are consistent with librosa.core.spectrum._spectrogram
+    waveform = torch.istft(
+        input=spectrogram,
+        n_fft=n_fft,
+        hop_length=hop_length,
+        win_length=win_length,
+        window=window,
+        center=center,
+        normalized=False,
+        onesided=onesided,
+        length=length + 2 * pad if length is not None else None,
+        return_complex=False,
+    )
+
+    if pad > 0:
+        # remove padding from front and back
+        waveform = waveform[:, pad:-pad]
```

Comment:
here, `pad` is applied when it's larger than 0. in the call to `torch.istft` above, it's applied when `length` is not `None`. is this discrepancy expected?

Reply:
@nateanl Is this resolved? The current code looks okay to me. My understanding is that `pad` is effective only when `length` is not `None`.